pandas/_libs/tslibs/np_datetime.pyx

```diff
@@ -33,7 +33,7 @@ cdef extern from "src/datetime/np_datetime.h":
     npy_datetimestruct _NS_MIN_DTS, _NS_MAX_DTS
 
 cdef extern from "src/datetime/np_datetime_strings.h":
-    int parse_iso_8601_datetime(char *str, int len,
+    int parse_iso_8601_datetime(const char *str, int len,
```

Comment:
just for my edification, what do we gain from this?

Reply:
This removes some warnings, and it is also more correct - ideally any parameter that function doesn't change should be passed as `const` so:
1. compiler could generate more optimal code, and
2. it would be more apparent from function signature that the function won't change this parameter :)

Reply:
This is the way to say that the function does not change this string.
Also, when I try to transfer `const char*` in this function without `const` for `str` variable then in complilation stage appeared: 
`error: invalid conversion from ‘const char*’ to ‘char*’ [-fpermissive]`